server/neptune/workflows/internal/deploy/revision/queue/pull_rebaser.go

```diff
@@ -0,0 +1,128 @@
+package queue
+
+import (
+	"context"
+	"time"
+
+	"github.com/docker/docker/pkg/fileutils"
+	"github.com/pkg/errors"
+	key "github.com/runatlantis/atlantis/server/neptune/context"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/github"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/terraform"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/config/logger"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/metrics"
+	"go.temporal.io/sdk/workflow"
+)
+
+// GH API Ratelimit resets every hour, 2 hours should be enough for the GH API call failures to autoresolve
+const RebaseGithubScheduleToCloseTimeout = 2 * time.Hour
+const RebaseTaskQueue = "rebase"
+
+type rebaseActivities interface {
+	GithubListOpenPRs(ctx context.Context, request activities.ListOpenPRsRequest) (activities.ListOpenPRsResponse, error)
+	GithubListModifiedFiles(ctx context.Context, request activities.ListModifiedFilesRequest) (activities.ListModifiedFilesResponse, error)
+	SetPRRevision(ctx context.Context, request activities.SetPRRevisionRequest) (activities.SetPRRevisionResponse, error)
+}
+
+type PullRebaser struct {
+	RebaseActivites rebaseActivities
+}
+
+// Called async in the deploy workflow after a deployment is complete
+func (p *PullRebaser) RebaseOpenPRsForRoot(ctx workflow.Context, repo github.Repo, root terraform.Root, scope metrics.Scope) error {
+
+	// custom tq for rebase activities to allow flow control and limit GH API calls per hour
+	ctx = workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+		TaskQueue:           RebaseTaskQueue,
+		StartToCloseTimeout: 10 * time.Second,
```

Comment:
Can you explain the benefits of the a 10 second StartToCloseTimeout for me?

Reply:
According to the docs, `StartToCloseTimeout` should be as long as the longest possible execution of the activity body. Since we only make GH API calls here, I configured it similar to how we've [configured](https://github.com/lyft/atlantis/blob/d7493e19175c2d29bd1e688549d537fc40138513/server/neptune/workflows/internal/deploy/workflow.go#L57) the deploy workflow (5 seconds).